Fixes to issues discussed in #53 and few more

RamTMO opened the pull request:
#### What type of PR is this?
 
* bug
 


#### What this PR does / why we need it:

It addresses the issues discussed in #53 along with few more minor fixes


#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #53 

#### Special notes for reviewers:



#### Changelog input

```
It addresses the issues discussed in #53 along with few more minor fixes

```

#### Additional documentation 

This section can be blank.



```
docs

```

jlurien reviewed (dismissed):
LGTM, please fix linter error

hdamker commented:
There is minor issue within line 1523: 
* uses single quotes '#/components/headers/x-correlator' while all other references use double quotes

hdamker commented on code/API_definitions/qos-booking-and-assignment.yaml:
```diff
@@ -276,7 +275,7 @@ paths:
                   $ref: "#/components/examples/BOOKING_CANCELLED"
         "202":
-          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `SCHEDULED`
+          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `SUCCESSFUL` and `statusInfo` will be `DELETE_REQUESTED`
```
```suggestion
          description: Deletion request accepted to be processed. It applies for an async deletion process. `status` in the response will be `PENDING` and `statusInfo` will be `DELETE_REQUESTED`
```

The example CANCELLATION_PENDING has the correct status `PENDING`.

hdamker reviewed (changes requested):
Have checked that all issues from #53 are addressed, found one point which need to be fixed.

hdamker reviewed (approved):
LGTM.

https://github.com/camaraproject/QoSBooking/pull/55#issuecomment-3242374705 is not addressed, but is just about style and consistency, so not functional relevant.

RamTMO commented:
> LGTM so far except linting error.
> 
> @RamTMO, could you fix the linting error? I found that cases where space were inserted at the end of the last line often happened. So I'd appreciated if you could be take care of it from now and onwards. Thank you.

Fixed this; fixed few others.

hdamker commented on code/API_definitions/qos-booking-and-assignment.yaml:
```diff
@@ -723,14 +722,15 @@ components:
         sink:
           type: string
           format: uri
+          pattern: "^https:\/\/.+$"
```
The pattern `^https:\/\/.+$` should use no quotes (as in Commonalities) or single quotes, but not double quotes (with double quotes the `\` will be interpreted in YAML as escape, changing the regex.

Let's align with commonalities and keep this pattern unquoted:

```suggestion
          pattern: ^https:\/\/.+$
```

hdamker commented on code/API_definitions/qos-booking-and-assignment.yaml:
```diff
@@ -1155,7 +1155,7 @@ components:
       description: A public identifier addressing a telephone subscription. In mobile networks it corresponds to the MSISDN (Mobile Station International Subscriber Directory Number). In order to be globally unique it has to be formatted in international format, according to E.164 standard, prefixed with '+'.
       type: string
-      pattern: '^\+[1-9][0-9]{4,14}$'
+      pattern: "^\+[1-9][0-9]{4,14}$"
```
The pattern: `'^\+[1-9][0-9]{4,14}$'` is single quoted in Commonalities to keep it literally, and must not be double quoted (see the syntax error by YAML lint). Please keep it like in Commonalities:

```suggestion
      pattern: '^\+[1-9][0-9]{4,14}$'
```

hdamker commented on code/API_definitions/qos-booking-and-assignment.yaml:
```diff
@@ -1310,7 +1310,7 @@ components:
     XCorrelator:
       type: string
-      pattern: ^[a-zA-Z0-9-_:;.\/<>{}]{0,256}$
+      pattern: "^[a-zA-Z0-9-_:;.\/<>{}]{0,256}$"
```
The pattern `^[a-zA-Z0-9-_:;.\/<>{}]{0,256}$` should use no quotes (as in Commonalities) or single quotes, but not double quotes (with double quotes the \ will be interpreted in YAML as escape, changing the regex)

Let's align with commonalities and keep this pattern unquoted:

```suggestion
      pattern: ^[a-zA-Z0-9-_:;.\/<>{}]{0,256}$
```

hdamker reviewed (changes requested):
@RamTMO Unfortunately you did too much within the last commit - I asked to change the quotes only in one instance. 

In YAML the differences between single and double quotes are important, see https://dev.to/darkmavis1980/single-quotes-vs-double-quotes-in-yaml-how-they-differ-from-javascript-and-typescript-20hj for background.

My suggestion is to use the pattern exactly as they are defined in the files in Commonalities.

RamTMO commented:
@hdamker accepted the changes.  It is interesting to understand the behavior of this syntax.  Thanks for pointing out.

Masa8106 commented on code/API_definitions/qos-booking-and-assignment.yaml:
```diff
@@ -1155,7 +1155,7 @@ components:
       description: A public identifier addressing a telephone subscription. In mobile networks it corresponds to the MSISDN (Mobile Station International Subscriber Directory Number). In order to be globally unique it has to be formatted in international format, according to E.164 standard, prefixed with '+'.
       type: string
-      pattern: '^\+[1-9][0-9]{4,14}$'
+      pattern: "^\+[1-9][0-9]{4,14}$"
```
This suggestion has already implemented.

Masa8106 commented:
@RamTMO, I don't think you need to do anything. I think we need an approval from @hdamker who requested changes.
@hdamker, may I ask you to review it?
